server/embed/config_test.go

```diff
@@ -429,3 +430,80 @@ func TestLogRotation(t *testing.T) {
 		})
 	}
 }
+
+func TestTLSVersionMinMax(t *testing.T) {
+	tests := []struct {
+		name                  string
+		config                Config
+		expectError           bool
+		expectedMinTLSVersion uint16
+		expectedMaxTLSVersion uint16
+	}{
+		{
+			name: "Minimum TLS version is set",
+			config: Config{
+				MinVersion: "TLS13",
+			},
+			expectedMinTLSVersion: tls.VersionTLS13,
+			expectedMaxTLSVersion: 0,
+		},
+		{
+			name: "Maximum TLS version is set",
+			config: Config{
+				MaxVersion: "TLS12",
+			},
+			expectedMinTLSVersion: 0,
+			expectedMaxTLSVersion: tls.VersionTLS12,
+		},
+		{
+			name: "Minimum and Maximum TLS versions are set",
+			config: Config{
+				MinVersion: "TLS13",
+				MaxVersion: "TLS13",
+			},
+			expectedMinTLSVersion: tls.VersionTLS13,
+			expectedMaxTLSVersion: tls.VersionTLS13,
+		},
+		{
+			name: "Minimum and Maximum TLS versions are set in reverse order",
+			config: Config{
+				MinVersion: "TLS13",
+				MaxVersion: "TLS12",
+			},
+			expectError: true,
+		},
+		{
+			name: "Invalid minimum TLS version",
+			config: Config{
+				MinVersion: "invalid version",
+			},
+			expectError: true,
+		},
+		{
+			name: "Invalid maximum TLS version",
+			config: Config{
+				MaxVersion: "invalid version",
+			},
+			expectError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := updateMinMaxVersions(&tt.config.PeerTLSInfo, tt.config.MinVersion, tt.config.MaxVersion)
+			if err != nil && !tt.expectError {
```

Comment:
Note that it doesn't cover the case `err == nil && tt.expectError`.

```suggestion
			if (err != nil) != tt.expectError {
```